include/clang/AST/Decl.h

```diff
@@ -731,6 +734,25 @@ class DeclaratorDecl : public ValueDecl {
 
   friend class ASTDeclReader;
   friend class ASTDeclWriter;
+
+  // Checked C bounds information
+
+  /// \brief Return true if this declaration has bounds declared for it.
+  bool hasBoundsExpr() const;
+
+  /// \brief The declared bounds for this declaration. For function
+  /// declarations, this is the return bounds of the function. Null if no
+  /// bounds have been declared.
+  const BoundsExpr *getBoundsExpr() const {
```

Comment:
>getBoundsExpr [](start = 20, length = 13)

Does clang have a general standard for how to indicate that certain members in base classes may not be applicable to certain derived classes?  Would it, say, be appropriate to add an assertion on setBoundsExpr() that it's only being set on an object on which it makes sense?